File: RestaurantApp/RestaurantAppServer/Domain/Entities/Order.cs
using System.ComponentModel.DataAnnotations;
using Domain.Enums;

namespace Domain.Entities;

public class Order
{
    [Key]
    public int Id { get; set; }
    public int UserId { get; set; }
    public User User { get; set; }
    public ICollection<OrderItem> OrderItems { get; set; }
    public OrderStatus Status { get; set; } 
    public decimal TotalAmount { get; set; }
}

File: RestaurantApp/RestaurantApp/RestaurantApp/Models/Order.swift
//
//  Order.swift
//  RestaurantApp
//
//  Created by Азат Зиганшин on 07.05.2024.
//

import Foundation

struct Order: Codable {
    let id: Int
    let userId: Int
    let user: User?
    let orderItems: [OrderItem]?
    let status: OrderStatus
    let totalAmount: Decimal
}
